v1/services/storage/store.go

```diff
@@ -650,27 +648,145 @@ func (s *Store) ReadSeriesCardinality(ctx context.Context, req *datatypes.ReadSe
 		return nil, err
 	}
 
-	if req.Range.Start == 0 {
-		req.Range.Start = models.MinNanoTime
+	db, rp, start, end, err := s.validateArgs(source.OrgID, source.BucketID, req.Range.Start, req.Range.End)
+	if err != nil {
+		return nil, err
+	}
+
+	sgs, err := s.MetaClient.ShardGroupsByTimeRange(db, rp, time.Unix(0, start), time.Unix(0, end))
+	if err != nil {
+		return nil, err
+	}
+
+	if len(sgs) == 0 {
+		return cursors.NewInt64SliceIterator([]int64{0}), nil
+	}
+
+	var expr influxql.Expr
+	if root := req.Predicate.GetRoot(); root != nil {
+		expr, err = reads.NodeToExpr(root, measurementRemap)
+		if err != nil {
+			return nil, err
+		}
+
+		if found := reads.HasFieldValueKey(expr); found {
+			return nil, errors.New("field values unsupported")
```

Comment:
Will this error get wrapped up with more prefixes before being shown to the user? If not, I think it'd be good to add a little more detail (i.e. `filtering on field values is not supported in cardinality predicates`)

Reply:
It doesn't really get wrapped in anything helpful so I updated the error message as you suggest. 

No matter what, it currently gives the rather unhelpful `Error: failed to execute query: 500 Internal Server Error: An internal error has occurred` message, but this should be improved by https://github.com/influxdata/influxdb/pull/22448

Reply:
Not anymore - now it's just a pointer to the logs again.

Ideally at some point this error would translate to something user-visible. (i.e. would be translated to an influxdb.Error)

Reply:
Ah I see, #22448 will at least log the error message though, so that's something.